wallet/src/de/schildbach/wallet/ui/CheckPinDialog.kt

```diff
@@ -52,44 +43,21 @@ import javax.inject.Inject
 
 @AndroidEntryPoint
 open class CheckPinDialog(
-    private val onSuccessOrDismiss: ((String?) -> Unit)?
+    private var onSuccessOrDismiss: ((String?) -> Unit)?
 ) : DialogFragment() {
 
     companion object {
 
         internal val FRAGMENT_TAG = CheckPinDialog::class.java.simpleName
         private val log = LoggerFactory.getLogger(CheckPinDialog::class.java)
 
-        internal const val ARG_REQUEST_CODE = "arg_request_code"
-        internal const val ARG_PIN_ONLY = "arg_pin_only"
-
-        private fun showDialog(checkPinDialog: CheckPinDialog, activity: FragmentActivity, requestCode: Int = 0, pinOnly: Boolean = false) {
-            val controller = PinRetryController.getInstance()
-
-            if (controller.isLocked) {
```

Comment:
`isLocked` check moved to `onCreateView` so that we can get rid of the `getInstance` call.